File: algorithms/1_easy/M/03168_minimum_number_of_chairs_in_a_waiting_room/solution.py
class Solution:
    def minimumChairs(self, s: str) -> int:

        max_chairs = 0
        balance = 0

        for ch in s:
            if ch == 'E':
                balance += 1
                max_chairs = max(max_chairs, balance)
            else:
                balance -= 1

        return max_chairs

File: algorithms/1_easy/M/03168_minimum_number_of_chairs_in_a_waiting_room/solution.cpp
class Solution {
public:
    int minimumChairs(string s) {

        int max{0};
        int balance{0};

        for (const char &ch: s) {
            if (ch == 'E') {
                balance += 1;
                max = std::max(max, balance);
            } else {
                balance -= 1;
            }
        }

        return max;
    }
};
